demos/04-convo/main.scm

```diff
@@ -594,15 +594,44 @@
   (define parent (vector-ref model 0))
   (define message (vector-ref model 1))
   (define uid (make-uid))
+  (set! db (nstore-add db space (vector uid "parent" parent)))
+  (set! db (nstore-add db space (vector uid "message" message)))
+  uid)
 
+(define (new-reply! parent text)
+  (define uid (make-uid))
   (set! db (nstore-add db space (vector uid "parent" parent)))
-  (set! db (nstore-add db space (vector uid "message" message))))
+  (set! db (nstore-add db space (vector uid "message" text))))
 
 (define (on-key-press model event)
   (let* ((key (webui-event-key event)))
     (if (and (string=? key "Enter") (not (string=? "" (vector-ref model 1))))
-        (begin (new-message! model) (vector-set! model 1 "") model)
-        model)))
+      (let ((msg-uid (new-message! model)))
+        (begin
+          (vector-set! model 1 "")
+          (schedule-a-mock-reply msg-uid)
```

Comment:
eventually the `schedule-a-mock-reply` will look like `(mock-reply msg (then change) (err change))` where then look like:

```scheme
(define (then change)
  (lambda (translation)
    (change (lambda (model) (new-message! translation) model)))
```

Or something along those lines...